### PR TITLE
Lesson15

### PR DESCRIPTION
**Pull request by DesoPy**

### CODE REVIEW

File: Lesson15/hw_lesson15.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+class Product:
+
+    c = 0
+
+    def __init__(self, types: str, name: str, price: float):
+        self.types = types
+        self.name = name
+        self.price = price
+        self.__class__.c += 1
```
Comment (fadich) [inline Lesson15/hw_lesson15.py]: Redutant value.

File: Lesson15/hw_lesson15.py [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        return self.types + ', ' + self.name + ', ' + str(self.price)
+
+
+class ProductStore:
+
+    def __init__(self):
+        self.goods = dict()
+
+    def add(self, product, amount):
+        self.goods[product.name] = [product.types, product.name, round(product.price * 1.3, 2), amount]
```
Comment (fadich) [inline Lesson15/hw_lesson15.py]: What if I add the same product twice?
```
store.add("Sushi", 100)
store.add("Sushi", 1)

store.sell("Sushi", 101)
```

File: Lesson15/hw_lesson15.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        return tuple(self.goods[product.name])
+
+    def set_discount(self, identifier, percent):
+        a = self.goods[identifier]
+        a.append(f'{percent} %')
+        self.goods.update({self.goods.get(self, identifier): a})
```
Comment (fadich) [inline Lesson15/hw_lesson15.py]:
```suggestion
        self.goods.update({self.goods.get(identifier): a})
```

File: Lesson15/hw_lesson15.py [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+        return self.goods
+
+    def sell(self, product_name, amount):
+        a = self.goods[product_name]
+        a[3] = a[3] - amount
+        self.goods.update({self.goods.get(self, product_name): a})
+        return (product_name, self.goods[product_name])
+
+    @staticmethod
+    def get_income(cls):
```
Comment (fadich) [inline Lesson15/hw_lesson15.py]: We just should make a property with `money_earned_total` and increase it on selling the product. There is a typo in the task description... ;)

File: Lesson15/hw_lesson15.py [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+        self.msg = super().__init__()
+        error_message = msg.__name__
+        text_error = msg.__doc__
+        date_error = datetime.now()
+        with open('../Lesson15/logs.txt', 'a') as file:
+            file.write(f'Date error: {date_error}\n name error: {error_message}\n text error: {text_error}\n')
+            file.write('#' * 100 + '\n')
+
+
+assert CustomException(TypeError)
```
Comment (fadich) [inline Lesson15/hw_lesson15.py:225]: The task made correctly, just incorrect usage. Try to use it this way:
```
raise CustomException("message")
```

**Review by fadich — APPROVED** [review]
There are some improvement suggestions, please, fix and merge afterwards.

File: Lesson15/hw_lesson15.py (hunk 1 above)
Comment (DesoPy) [reply Lesson15/hw_lesson15.py]: Changed.

File: Lesson15/hw_lesson15.py (hunk 2 above)
Comment (DesoPy) [reply Lesson15/hw_lesson15.py]: Error, changed.

File: Lesson15/hw_lesson15.py (hunk 3 above)
Comment (DesoPy) [reply Lesson15/hw_lesson15.py]: Changed.

File: Lesson15/hw_lesson15.py (hunk 4 above)
Comment (DesoPy) [reply Lesson15/hw_lesson15.py]: Changed.

File: Lesson15/hw_lesson15.py (hunk 5 above)
Comment (DesoPy) [reply Lesson15/hw_lesson15.py:225]: Try, changed, if I understood correctly.

**Pull request merged**